### PR TITLE
Expose the screen sharing source id

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
Would it be possible to split the jshint removal and the feature change itself into separate commits?

File: modules/RTC/JitsiLocalTrack.js [hunk 1]
```diff
@@ -15,696 +15,697 @@ import VideoType from '../../service/RTC/VideoType';
...
-    if (this._noDataFromSourceTimeout) {
-        clearTimeout(this._noDataFromSourceTimeout);
-        this._noDataFromSourceTimeout = null;
+    /**
+     * Sets the stream property of JitsiLocalTrack object and sets all stored
+     * handlers to it.
+     * @param {MediaStream} stream the new stream.
+     */
+    _setStream(stream) {
+        JitsiTrack.prototype._setStream.call(this, stream);
```
Comment (virtuacoplenny) [inline modules/RTC/JitsiLocalTrack.js]: I think you're also changing the functions a bit for es6. Am I correct assuming so? Could this be changed to a super._setStream call?

File: modules/RTC/JitsiRemoteTrack.js [hunk 2]
```diff
@@ -40,222 +44,236 @@ export default function JitsiRemoteTrack(
...
+            this.hasBeenMuted = true;
+        }
+
+        // we can have a fake video stream
+        if (this.stream) {
+            this.stream.muted = value;
+        }
+
+        this.muted = value;
+        this.eventEmitter.emit(JitsiTrackEvents.TRACK_MUTE_CHANGED, this);
```
Comment (virtuacoplenny) [inline modules/RTC/JitsiRemoteTrack.js]: Do these these have a reference to this.eventEmitter? Would these calls be to this.emit?

File: .jshintignore [hunk 3]
```diff
@@ -1,11 +0,0 @@
...
-# source code.
-doc/
-node_modules/
-lib-jitsi-meet.js
-lib-jitsi-meet.js.map
-lib-jitsi-meet.min.js
-
-# The following are checked by ESLint and take advantage of advanced language
-# features such as Facebook Flow which are not supported by JSHint.
-flow-typed/
```
Comment (virtuacoplenny) [inline .jshintignore]: Is jshintrc still needed?

File: modules/RTC/JitsiLocalTrack.js [hunk 4]
```diff
@@ -46,28 +46,34 @@ function createMuteUnmutePromise(track, mute) {
...
+     * @param trackInfo.stream WebRTC MediaStream, parent of the track
+     * @param trackInfo.track underlying WebRTC MediaStreamTrack for new
+     * JitsiRemoteTrack
+     * @param trackInfo.mediaType the MediaType of the JitsiRemoteTrack
+     * @param trackInfo.videoType the VideoType of the JitsiRemoteTrack
+     * @param trackInfo.resolution the video resolution if it's a video track
+     * @param trackInfo.deviceId the ID of the local device for this track
+     * @param trackInfo.facingMode the camera facing mode used in getUserMedia
+     * call
      * @constructor
```
Comment (virtuacoplenny) [inline modules/RTC/JitsiLocalTrack.js]: You might want to add sourceId to the doc.

File: modules/RTC/RTCUtils.js [hunk 5]
```diff
@@ -628,14 +628,16 @@ function handleLocalStream(streams, resolution) {
             videoStream = streams.video;
         }
 
-        // Again, different choices on different types of browser.
-        desktopStream = streams.desktopStream || streams.desktop;
+        desktopStream = streams.desktop;
```
Comment (virtuacoplenny) [inline modules/RTC/RTCUtils.js]: Why was "streams.desktopStream" needed in the past?

File: modules/RTC/JitsiRemoteTrack.js (hunk 2 above)
Comment (hristoterezov) [inline modules/RTC/JitsiRemoteTrack.js]: No. Will fix it!

File: .jshintignore (hunk 3 above)
Comment (hristoterezov) [inline .jshintignore]: Not really.

File: modules/RTC/RTCUtils.js (hunk 5 above)
Comment (hristoterezov) [inline modules/RTC/RTCUtils.js]: We were naming the prop `desktopStream` for some browsers and for others `desktop`. I don't believe this was intentional.

**Comment by hristoterezov** [comment]
@virtuacoplenny I think I addressed all comments!

File: package.json [hunk 6]
```diff
@@ -39,7 +39,6 @@
     "eslint-plugin-import": "2.3.0",
     "flow-bin": "0.38.0",
     "jasmine-core": "2.5.2",
-    "jshint": "2.9.4",
```
Comment (virtuacoplenny) [inline package.json]: I noticed some inline jshint disabling still left behind. Maybe it's good to just remove those now?

Comment (hristoterezov) [inline package.json]: Fixed!

Comment (virtuacoplenny) [inline package.json]: I still see some references in adapter.screenshare. Were those part of the third-party library?

Comment (hristoterezov) [inline package.json]: Yes. It's provided by Temasys. We are updating it from time to time. It's interesting why it is part of our source code but unfortunately I don't know. 

Anyway this is whole different story :)

**Pull request merged**